Jenkinsfile

```diff
@@ -16,17 +16,30 @@ node {
 
 properties([
     disableConcurrentBuilds(),
-    pipelineTriggers(devel ? [] : [cron("H/30 * * * *")])
+    pipelineTriggers(devel ? [] : [cron("H/30 * * * *")]),
+    parameters([
+      choice(name: 'STREAM',
+             // XXX: Just pretend we're the testing stream for now... in
+             // reality, we're closer to what "bodhi-updates" will be. Though the
+             // testing stream is the main stream.
+             choices: ['testing' /*, 'stable', 'testing-devel', 'bodhi-updates', etc... */ ],
+             description: 'Fedora CoreOS stream to build',
+             required: true)
+    ])
 ])
 
 podTemplate(cloud: 'openshift', label: 'coreos-assembler', yaml: pod, defaultContainer: 'jnlp') {
     node('coreos-assembler') { container('coreos-assembler') {
 
         stage('Init') {
             utils.shwrap("""
-            if [ ! -d src/config ]; then
-                coreos-assembler init https://github.com/coreos/fedora-coreos-config
-            fi
+            # just always restart from scratch in case it's a devel pipeline
+            # and it changed source url or ref; this info also makes it into
+            # the build metadata through cosa reading the origin remote
+            rm -rf src/config
+
+            # in the future, the stream will dictate the branch in the prod path
+            coreos-assembler init --force https://github.com/coreos/fedora-coreos-config
```

Comment:
I wonder if the URL should be templatized

Reply:
Yup, that's coming up in a next patch. :)